streaming/database.js

```diff
@@ -116,13 +116,44 @@ let pool;
 /**
  *
  * @param {pg.PoolConfig} config
+ * @param {string} environment
+ * @param {import('pino').Logger} logger
  * @returns {pg.Pool}
  */
-export function getPool(config) {
+export function getPool(config, environment, logger) {
   if (pool) {
     return pool;
   }
 
   pool = new pg.Pool(config);
+
+  // Setup logging on pool.query and client.query for checked out clients:
+  // This is taken from: https://node-postgres.com/guides/project-structure
+  if (environment === 'development') {
```

Comment:
Could you maybe move this in a separate PR?

Reply:
Moved to #32945